src/main/java/uk/gov/hmcts/darts/common/repository/ExternalObjectDirectoryRepository.java

```diff
@@ -62,16 +62,17 @@ List<ExternalObjectDirectoryEntity> findByMediaStatusAndType(MediaEntity media,
     List<ExternalObjectDirectoryEntity> findByTranscriptionDocumentEntityAndExternalLocationType(TranscriptionDocumentEntity transcriptionDocument,
                                                                                                  ExternalLocationTypeEntity externalLocationType);
 
-    @Query(
-        """
-                SELECT eod.media.id FROM ExternalObjectDirectoryEntity eod
-                WHERE eod.media.id in :mediaIdList
-                AND eod.status = :status
-                AND eod.externalLocationType = :externalLocationType
-            """
+    @Query(value = """
+                SELECT eod.med_id 
+                FROM darts.external_object_directory eod
+                WHERE eod.med_id in (:mediaIdList)
+                AND eod.ors_id = :status
+                AND eod.elt_id = :externalLocationType
+            """,
+        nativeQuery = true
```

Comment:
@tonymort / @mario-paniccia  is there a downside to using native sql?
does it get validated at start-up etc?
Is it as maintainable?

Reply:
we have some other native queries already, so I assumed it's accepted. 
The alternative is spending more time to figure out why that weird SQL is being generated from the JPQL. Perhaps I can try replacing `eod.media.id` with `eod.media` and passing a List<MediaEntity>. 
I didn't go with that solution because is a bigger change, and we don't have much time on these tickets

Reply:
i've been doing some digging, it looks like when the original query:-
```sql
SELECT eod.media.id FROM ExternalObjectDirectoryEntity eod
WHERE eod.media.id in ( :mediaIdList )
AND eod.status = :status
AND eod.externalLocationType = :externalLocationType
```

was provided with an *empty* list, it produced this SQL:-
```sql
select
        eode1_0.med_id 
    from
        darts.external_object_directory eode1_0 
    where
        (
            1 = case 
                when eode1_0.med_id is not null 
                    then 0 
            end
        ) 
        and eode1_0.ors_id=2
        and eode1_0.elt_id=1;
        ```
        
        Which seemed to be really badly performing.
        But when provided with a populated list, it produces this:-
        ```sql
    select
        eode1_0.med_id 
    from
        darts.external_object_directory eode1_0 
    where
        eode1_0.med_id in (1) 
        and eode1_0.ors_id=2
        and eode1_0.elt_id=1;
        ```
        
        which performs really well.
        
so i think it may be best to bypass calling this method if the list is empty.

Reply:
nice finding @cmagowan-hmcts, what you suggested is the best solution then. I'll make the changes cheers